main.go

```diff
@@ -200,7 +227,11 @@ func mergeFilesInHierarchy(hierarchy []string, fileFilter string, outputFile str
 		"path": outputFile,
 	}).Info("Writing output file")
 	yamlDoc, err := yaml.Marshal(&data)
-	err = ioutil.WriteFile(outputFile, yamlDoc, 0660)
+	yamlDocStr := string(yamlDoc)
+	if !skipEnvVarContent {
+		yamlDocStr = replaceEnvironmentVariables(string(yamlDocStr), failMissingEnvVar)
```

Comment:
Since yamlDocStr is already converted to string, is there any point to do it again?

```suggestion
		yamlDocStr = replaceEnvironmentVariables(yamlDocStr, failMissingEnvVar)
```